source/monitoring/cluster-monitoring.txt

```diff
@@ -0,0 +1,135 @@
+.. _php-cluster-monitoring:
+
+==================
+Cluster Monitoring
+==================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, server, topology
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+This guide shows you how to use the {+php-library+} to monitor topology
+events in a MongoDB instance, replica set, or sharded cluster. The
+driver creates topology events, also known as Server Discovery and
+Monitoring (**SDAM**) events, when there are any changes in the state of the
+MongoDB instance or cluster that you are connected to.
+
+You might use information about topology events in your
+application to understand cluster changes, assess cluster health, or
+perform capacity planning.
+
+Subscribe to Events
+-------------------
+
+You can access details about SDAM events by subscribing to them
+in your application. To subscribe to any event, create a class that
+implements the ``MongoDB\Driver\Monitoring\SDAMSubscriber`` interface,
+then use the ``MongoDB\Client::addSubscriber()`` method to register the
+event subscriber with your ``MongoDB\Client`` instance.
+
+The following code creates the ``MySubscriber`` class that implements
+``SDAMSubscriber`` to output a message when a ``ServerOpeningEvent`` is
+generated by the server:
+
+.. literalinclude:: /includes/monitoring/sdam.php
+   :start-after: start-mysubscriber
+   :end-before: end-mysubscriber
+   :language: php
+   :copyable:
+   :dedent:
+
+.. note::
+
+   As shown in the preceding code, you must implement all of the methods
+   of ``SDAMSubscriber``, even for events you are not subscribing to.
+   You can implement these methods with empty bodies so that the library
+   does not generate any messages for these events.
+
+Then, the following code uses the ``addSubscriber()`` method to register
+``MySubscriber`` with the client:
```

Comment:
I: I like the use of "then" to signify a subsequent step, but I would tie it to a user step instead of the code example. Something like:

```suggestion
Then, use the ``addSubscriber()`` method to register
``MySubscriber`` with the client, as shown in the following code:
```